src/lib/containers/widgets/query-filter/QueryFilter.tsx

```diff
@@ -141,19 +143,29 @@ export const QueryFilter: React.FunctionComponent<QueryFilterProps> = ({
   const { data, isLoadingNewBundle, getLastQueryRequest, executeQueryRequest } =
     useQueryContext()
 
+  const facets = data?.facets
+
+  let shownChips: string[]
+  if (facetsToFilter == null) {
+    shownChips = facets?.map(facet => facet.columnName) ?? []
+  } else {
+    shownChips = facetsToFilter
+  }
+  const [facetFiltersShown, setFacetFiltersShown] = React.useState<string[]>([])
   const { facetAliases, topLevelControlsState } = useQueryVisualizationContext()
+  const { showFacetFilter } = topLevelControlsState
+
+  /**
+   * When the data facets change, reset the initially-selected chips
+   */
+  useDeepCompareEffectNoCheck(() => {
+    setFacetFiltersShown(shownChips.slice(0, 3))
```

Comment:
Please remove the `shouldStartCollapsed` variable elsewhere in this file and replace with `false`, so all filter controls are expanded